skimage/measure/_regionprops.py

```diff
@@ -358,22 +358,93 @@ def __eq__(self, other):
         return True
 
 
+def _props_to_dict(regions, properties=('label', 'bbox'), separator='-', always_include_label=True):
+    """Convert image regions properties into a dictionary
+
+    Parameters
+    ----------
+    regions : (N,) list
+        List of _RegionProperties objects as returned by :func:`regionprops`
+    properties : tuple, optional
+        properties which the output need to have
+    separator : str, optional
+        Each element of non-scalar properties and is not listed in object_columns will
+        be put into its own column, with the index of that element separated from
+        the name by this separator.
+    always_include_label : bool, optional
+        allows autmatic adding of label to the output
+
+
+    Output
+    ------
+    A dictionary with properties in accordance to the wanted properties
+
+    Examples
+    --------
+    >>> from skimage import data, util, measure
+    >>> image = data.coins() > 110
+    >>> label_image = measure.label(image, connectivity=image.ndim)
+    >>> propslist = regionprops(label_image, image)
+    >>> props = to_dict(propslist, properties=['label', 'inertia_tensor', 'inertia_tensor_eigvals'])
+    >>> props
+    <output here>
+
+    The resulting dictionary can be directly passed to pandas, if installed, to
+    obtain a clean DataFrame::
+
+    >>> import pandas as pd  # doctest: +SKIP
+    >>> data = pd.DataFrame(props)  # doctest: +SKIP
+    >>> data.head()  # doctest: +SKIP
+    <output of data.head()>
+
+
+"""
+
+    object_columns = {'image', 'coords', 'convex_image', 'filled_image',
+               'intensity_image'}
+
+    if always_include_label and 'label' not in properties:
+        properties = ('label',) + properties
+
+    out = {}
+    arr = len(regions) * [0]
+    for prop in properties:
+        r = regions[0][prop]
+        if np.isscalar(r) or prop in object_columns:
+            out[prop] = [regions[i][prop] for i in range(len(arr))]
+        else:
+            if isinstance(r, np.ndarray):
+                shape = r.shape
+            else:
+                shape = (len(r),)
+
+            for ind in np.ndindex(shape):
+                for k in range(len(arr)):
+                    arr[k] = regions[k][prop][ind if len(ind) > 1 else ind[0]]
+                out[separator.join(map(str, (prop,) + ind))] = arr[:]
+
+    return out
+
+
+def regionprops_table(label_image, intensity_image=None, cache=True,
+                      coordinates=None, properties=('label', 'bbox'), separator='-',
+                      always_include_label=True):
+    """
+    Refer to regionprops and _props_to_dict
```

Comment:
This is a public-facing function, so it needs a full docstring, even if there is repetition. Users can't see `_props_to_dict` easily.

Reply:
What does this function contribute that is not already covered by `to_dict`?

Reply:
@stefanv in an earlier review, I mentioned to @leGIT-bot that `to_dict` was not a great name for a public function, and that anyway people probably want a one-shot function rather than having to call `regionprops` and then `to_dict` (with a better name) on the result.

Reply:
OK, makes sense.  Good to keep both then, or turn to_dict into a private method?

Reply:
Yes, that was my recommendation, but @leGIT-bot has yet to implement that bit.